### PR TITLE
search bar,posts list and categories list added on home page

### PR DESCRIPTION
**Pull request by jyotigautam**
Home page created having 3 containers-header,category list and posts list .Furthermore,header is having searchbar component,category list is having category list item component and posts list with posts list item component.The category list item once clicked,lists the posts related to that particular category.

### CODE REVIEW

File: src/app/app.js [hunk 1]
```diff
@@ -1,10 +1,21 @@
 import React, { Component } from 'react';
+import CategoriesList from './home/container/categories_list';
+import Header from './home/container/header';
 
 export default class App extends Component {
+
+  onSearched(keyword) {
+    console.log(keyword);
```
Comment (ashish173) [inline src/app/app.js]: @jyotigautam remove console logs.

File: src/app/home/component/searchbar.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+const Searchbar = (props) => {
+		return (
+				<div>
+						<input
+								onKeyPress = { (event) => props.onSearched(event.target.value) }
+								className = "search-bar"/>
+				</div>
+		);
+}
+
```
Comment (ashish173) [inline src/app/home/component/searchbar.js]: Tab space is totally off in this.

File: src/app/home/container/categories_list.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      .props
+      .fetchCategories();
+
+    this.onClicked = this
+      .onClicked
+      .bind(this);
+    this.renderCategories = this
+      .renderCategories
+      .bind(this);
+  }
```
Comment (ashish173) [inline src/app/home/container/categories_list.js:24]: @jyotigautam the above statements can be in one line IMO @pkrawat1 @zeus999 @kartikjagdale thoughts?

File: src/app/home/container/categories_list.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  render() {
+    return (
+      <ul>
+        Categories:
+        { this
+          .props
+          .categories
+          .map(this.renderCategories) }
```
Comment (ashish173) [inline src/app/home/container/categories_list.js]: @jyotigautam same here for next line statements.

File: src/app/home/container/header.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+class Header extends Component {
+	onSearched(keyword) {
+    console.log(keyword);
+  }
+
+	render(){
+	  return (
+	   <div className = "header">
+	     TAKE A PLEDGE FOR YOUR BETTER FUTURE
+			 <Searchbar onSearched = { (keyword) => this.onSearched(keyword) }/>
```
Comment (ashish173) [inline src/app/home/container/header.js:13]: @jyotigautam indentation is off I guess in this file.

File: src/app/home/services/home_service.js [hunk 6]
```diff
@@ -2,28 +2,58 @@ var posts = [
...
 
 export function getPosts() {
   return posts;
-}
\ No newline at end of file
+
+}
+
+export function getPostById(id) {
+  return posts[id-1];
```
Comment (ashish173) [inline src/app/home/services/home_service.js:47]: @jyotigautam please explain this logic.

File: src/app/home/services/home_service.js [hunk 7]
```diff
@@ -2,28 +2,58 @@ var posts = [
...
+}
+
+export function getCategories() {
+  return categories;
+}
+
+export function getCategorylist(category_id){
+  var categories_list = posts.filter(function (post) {
+    return post.category_id == category_id
```
Comment (ashish173) [inline src/app/home/services/home_service.js:56]: @jyotigautam use === in JS or TS, you could read more about it.

File: src/app/home/component/searchbar.js (hunk 2 above)
Comment (pkrawat1) [reply src/app/home/component/searchbar.js]: Ye kya hai 😤

File: src/store/posts/actions.js [hunk 8]
```diff
@@ -1,8 +1,25 @@
...
+}
+
+export function fetchpostsbyCategory(category_id) {
+  var posts_list = postService.getCategorylist(category_id);
+  return {type: FETCH_POSTS_BY_CATEGORY, payload: posts_list};
+}
+
+export function selectedPost(id){
+  var post = postService.getPostById(id);
+	return {
```
Comment (ashish173) [inline src/store/posts/actions.js:21]: @jyotigautam indentation is off again

File: src/store/posts/reducers.js [hunk 9]
```diff
@@ -1,18 +1,30 @@
...
+        ...state,
+        posts: action.payload
+      }
+    case 'FETCH_POSTS_BY_CATEGORY':
+      return {
+        ...state,
+        posts: action.payload
+      }
+    case 'FETCH_SELECTED_POST':
+		  return {
```
Comment (ashish173) [inline src/store/posts/reducers.js]: here as well.

File: src/style.css [hunk 10]
```diff
@@ -0,0 +1,40 @@
+.posts-list{
```
Comment (ashish173) [inline src/style.css:1]: @jyotigautam leave spaces throughout the file after the property name `background-color:#ffffff;` should be `background-color: #ffffff;`.

Also, don't use hardcoded colour hex create a scss variable. Actually let's switch to scss going forward.

@pkrawat1 @kartikjagdale @zeus999 thoughts ?

File: src/app/home/services/home_service.js (hunk 6 above)
Comment (jyotigautam) [reply src/app/home/services/home_service.js:47]: The post with id 1 or n is 0th or (n-1)th element in posts array and hence this logic.

File: src/app/home/container/categories_list.js (hunk 3 above)
Comment (chandradot99) [reply src/app/home/container/categories_list.js:24]: @jyotigautam since the statement is not too long, so it can be in one line.

**Pull request closed without merging**